core/trino-main/src/test/java/io/trino/sql/query/TestListagg.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.query;
+
+import io.trino.spi.TrinoException;
+import io.trino.sql.parser.ParsingException;
+import org.testcontainers.shaded.org.apache.commons.lang.StringUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.trino.spi.StandardErrorCode.EXCEEDED_FUNCTION_MEMORY_LIMIT;
+import static io.trino.spi.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestListagg
+{
+    private QueryAssertions assertions;
+
```

Comment:
For each of the tests in this class, add a variant involving `GROUP BY`. Currently, it's only testing the global aggregation case.

Reply:
I've added several new tests. 
I concentrated on the global aggregation before and left the grouping scenarios rather untested.

I've noticed that `DISTINCT` with `GROUP BY` don't work good together. This is a known limitation right?

Reply:
They should. Can you show me the query where they don't work?

Reply:
I must have misread the exception regarding the failure of the `DISTINCT` and `GROUP BY` test. I've added now a new test and it works as expected.